### PR TITLE
Make log levels button keyboard focusable and activable

### PR DESCRIPTION
**Pull request by Fenn-CS**
Improves accessibility by activating focus visible and making the log level button trigger-able from the keyboard as well.

Since both the text and the icons trigger the same action, both of them were wrapped in a button to make this easier.

Resolves: https://github.com/nextcloud/server/issues/36985

## Previews

🏚️ Before | 🏡 After
---|---
Screenshot from 2023-06-09 10-40-07 | Screenshot from 2023-06-09 10-40-38

### CODE REVIEW

File: src/Components/LogTable.js [hunk 1]
```diff
@@ -144,9 +144,12 @@ export class LogTable extends Component {
...
 
 		const levelHeader = (
-			<>{t('logreader', 'Level')}
+			<>
+			<button>
+			{t('logreader', 'Level')}
 				<span className={style['log-settings-toggle'] + ' icon-more'}>
 				</span>
+			</button>
 			</>
```
Comment (Antreesy) [inline src/Components/LogTable.js:153]:
```suggestion
			<button>
				{t('logreader', 'Level')}
				<span className={style['log-settings-toggle'] + ' icon-more'}>
				</span>
			</button>
```

Don't need a fragment there anymore, as everything is wrapped with button

File: src/Components/LogTable.css [hunk 2]
```diff
@@ -15,6 +15,26 @@
...
+            border: none !important;
+            background: none;
+            color: inherit;
+            border: none;
+            padding: 0;
+            font: inherit;
+            cursor: pointer;
+            outline: inherit;
+			border-radius: 5px;
+			width: 100%;
```
Comment (Antreesy) [inline src/Components/LogTable.css:30]:
```suggestion
            outline: inherit;
            border-radius: 5px;
            width: 100%;
```

File: src/Components/LogTable.css [hunk 3]
```diff
@@ -15,6 +15,26 @@
...
 		word-wrap: break-word;
 		border-bottom: 1px solid var(--color-border);
+
+		button {
+			display: flex;
+            border: none !important;
+            background: none;
+            color: inherit;
+            border: none;
+            padding: 0;
```
Comment (Antreesy) [inline src/Components/LogTable.css:25]: I'd also add a margin to align button to the top of the row
```suggestion
            margin: 0;
            padding: 0;
```

**Review by Antreesy — COMMENTED** [review]
Hi, sorry for a long delay!
Everthing seems fine and working, I left some comments where
Please also rebase and recompile the files, the have confilcts with master branch

**Comment by Fenn-CS** [comment]
Thanks @Antreesy resolved the mentioned issues, and manual compiled (before seeing you did the compile command already)

**Comment by Antreesy** [comment]
Looks like runner doesn't like manual build:
https://github.com/nextcloud/logreader/actions/runs/5332467137/jobs/9661680803?pr=887#step:7:5

File: src/Components/LogTable.css [hunk 4]
```diff
@@ -15,6 +15,27 @@
 		position: relative;
 		word-wrap: break-word;
 		border-bottom: 1px solid var(--color-border);
+
+		button {
+			display: flex;
+			border: none !important;
+			background: none;
```
Comment (Antreesy) [inline src/Components/LogTable.css:22]: Just noticed, that selector now affects all buttons in `<td>` also, so 'icon-clippy' is no longer visible on hover. Maybe move it to line 8?

File: src/Components/LogTable.css.d.ts [hunk 5]
```diff
@@ -1,30 +1,30 @@
...
+    active: string;
+    app: string;
+    column: string;
+    copy: string;
+    copyMenu: string;
+    empty: string;
+    highlight: string;
+    level: string;
+    level_1: string;
+    level_2: string;
```
Comment (Antreesy) [inline src/Components/LogTable.css.d.ts:12]: Please check that also: property 'focus-within' or 'focus-visible' was removed, but file still has changes due to indent (tabs or spaces?)

**Review by icewind1991 — APPROVED** [review]

**Comment by Fenn-CS** [comment]
@Antreesy 

- I moved the newly added button style in the `th` section 
- I remove the specific outline style added because the log levels action is now wrapped in a button which already has a style for that. 
    - See https://github.com/nextcloud/nextcloud-vue/issues/2824 and https://github.com/nextcloud/nextcloud-vue/pull/4176 where we are maintaining `--color-primary-element` for outlines.

    ### New look
     Screenshot from 2023-06-22 00-28-32

- This also fixes the issue with the clippy icon

File: src/Components/LogTable.css [hunk 6]
```diff
@@ -5,6 +5,21 @@
 
 	th {
 		color: var(--color-text-maxcontrast);
+
+		button {
+			display: flex;
+			border: none !important;
+			background: none;
+			color: inherit;
+			border: none;
```
Comment (Antreesy) [inline src/Components/LogTable.css:14]: Duplicated rule `border: none` - you added the same at L11

**Review by Antreesy — APPROVED** [review]
Apart from one comment, cool!

File: src/Components/LogTable.css (hunk 6 above)
Comment (Fenn-CS) [reply src/Components/LogTable.css:13]: thanks resolved.

**Pull request merged**